client/dashboard/sites/add-new-site/menu-item.tsx

```diff
@@ -0,0 +1,40 @@
+import {
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	Button,
+} from '@wordpress/components';
+import { Icon } from '@wordpress/icons';
+import React from 'react';
+
+type Props = {
+	children?: JSX.Element;
+	description: string;
+	title: string;
+	icon: JSX.Element;
+};
+
+function MenuItem( {
+	children,
+	description,
+	title,
+	icon,
+	...buttonProps
+}: Props & React.ComponentProps< typeof Button > ) {
+	return (
+		<Button { ...buttonProps } label={ title } style={ { height: 'auto', textAlign: 'left' } }>
+			<HStack alignment="start">
+				<div style={ { flexShrink: 0 } }>
+					<Icon className="dashboard-add-new-site__menu-item-icon" icon={ icon } />
+				</div>
+				<VStack>
+					<Text>{ title }</Text>
+					<Text variant="muted">{ description }</Text>
```

Comment:
Re avoiding the use of `preventWidows` as per the comment above, I'd suggest rendering Text `as="p"` here, so its contents get the default `text-wrap: pretty` applied to paragraphs. Also, given the structure it looks like a `p` is a better semantic match for this than a `span` which I believe is `Text`'s default element?